packages/sources/src/sourceFormRenderer/components/CardSelect.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Card, CardHeader, CardBody, FormGroup, Grid, GridItem, Bullseye } from '@patternfly/react-core';
+import { ServerIcon } from '@patternfly/react-icons';
+
+class CardSelect extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            selectedValues: this.props.input.value ? this.props.input.value : this.isMulti ? [] : undefined
+        };
+    }
+
+    isMulti = this.props.isMulti || this.props.multi;
+
+    isDisabled = this.props.isDisabled || this.props.isReadOnly;
+
+    change = () => this.props.input.onChange(this.state.selectedValues);
+
+    handleMulti = (value) => this.state.selectedValues.includes(value) ?
+        this.setState(({ selectedValues }) => ({ selectedValues: selectedValues.filter(valueSelect => valueSelect !== value) }), this.change)
+        : this.setState(({ selectedValues }) => ({ selectedValues: [ ...selectedValues, value ]}), this.change);
+
+    handleSingle = (value) => this.setState(() => ({ selectedValues: this.state.selectedValues === value ? undefined : value }), this.change);
+
+    handleClick = (value) => {this.isMulti ? this.handleMulti(value) : this.handleSingle(value);}
+
+    handleCompare = (value) => this.isMulti ? this.state.selectedValues.includes(value) : this.state.selectedValues === value;
+
+    onClick = value => {
+        if (this.isDisabled) {
+            return undefined;
+        }
+
+        this.handleClick(value);
+        this.props.input.onBlur();
+    }
+
+    handleKeyPress = (event, value) => {
+        if (event.key === 'Enter') {
+            this.onClick(value);
+        }
+    }
+
+    prepareCards = () => this.props.options.map(({ value, label }) => {
+        const { iconMapper, DefaultIcon } = this.props;
+
+        if (!value) {
```

Comment:
This condition should not be necessary, users should follow defined API. I would not be against actually throwing an error.

Reply:
It was made to add a option to easily share the options prop between Select (which supports undefined values) and this component.

Reply:
that makes sense :hurtrealbad: